packages/web/hooks/use-swap.ts

```diff
@@ -531,3 +519,96 @@ function useSwapAsset(
     isLoading: isLoading && !existingAsset,
   };
 }
+
+/** Iterates over available and identical routers and sends input to each one individually.
+ *  Results are reduced to best result by out amount.
+ *  Also returns the number of routers that have fetched and errored. */
+function useQueryRouterBestQuote(
+  input: RouterInputs["edge"]["quoteRouter"]["routeTokenOutGivenIn"],
+  enabled: boolean,
+  routerKeys = ["legacy", "sidecar", "tfm"] as RouterKey[]
+) {
+  const featureFlags = useFeatureFlags();
+  const availableRouterKeys: RouterKey[] = useMemo(
+    () =>
+      !featureFlags._isInitialized
+        ? []
+        : routerKeys.filter((key) =>
+            featureFlags.sidecarRouter ? true : key !== "sidecar"
+          ),
+    [featureFlags._isInitialized, featureFlags.sidecarRouter, routerKeys]
+  );
+
+  const trpcReact = createTRPCReact<AppRouter>();
+  const routerResults = trpcReact.useQueries((t) =>
+    availableRouterKeys.map((key) =>
+      t.edge.quoteRouter.routeTokenOutGivenIn(
+        {
+          ...input,
+          preferredRouter: key,
+        },
+        {
+          enabled: enabled && featureFlags._isInitialized,
+
+          // quotes should not be considered fresh for long, otherwise
+          // the gas simulation will fail due to slippage and the user would see errors
+          staleTime: 5_000,
+          cacheTime: 5_000,
+          // don't retry quote, just display the issue to user
+          retry: false,
```

Comment:
Thoughts on enabling retry only when the error is an unexpected (instanceof Error)?

Reply:
Is it possible to dynamically change that value from prior results?

Reply:
Yes! retry can be a [function](https://github.com/osmosis-labs/osmosis-frontend/blob/245c244b6d29dbfa9197edb9c27110154452cdaf/packages/web/modals/bridge-transfer-v2.tsx#L349-L356) that will receive the amount of failures and the last error.